django/pyrog/api/views.py

```diff
@@ -15,6 +15,20 @@ def get_serializer_class(self):
             return SourceSerializer
         return basic_serializers.SourceSerializer
 
+    def get_queryset(self):
+        """Limit visibility of sources."""
+        return self.queryset.filter(users=self.request.user)
+
+    def perform_create(self, serializer):
+        """Try to assign a owner to the new source."""
+
+        source = serializer.save()
+
+        # To give the source a owner, the request must be authenticated
+        if not self.request.user.is_anonymous:
```

Comment:
isn't the route already protected by authentication ? how can `self.request.user.is_anonymous` be True ?

Reply:
* in prod, users must be authenticated (enforce globally)
* in dev, users can be unauthenticated (and we should strive to maintain this for convenience). They can create resources.
* Currently we use the `SourceUser` (previously `AccessControl`) to restrict access to objects. Hence users are effectively limited to their sources, EVEN in dev. This means unauth users will not see any sources. The better way to deal with object permissions is to use a object-level permissions. I have found the django implementation of object-level permissions annoying. Having a permission framework like `open policy agent` could be better (framework/language agnostic, policy agnostic). With this, the development effort could be shared fastapi&django backends, kafka, HAPI etc.